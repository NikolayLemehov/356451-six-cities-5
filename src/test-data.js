// import {upperCaseFirst} from "./utils";

// const currentCityOffers = [
//   {
//     id: `1`,
//     img: [`url1`, `url2`, `url3`, `url4`, `url5`, `url6`],
//     smallImg: `small-url`,
//     title: `title`,
//     description: [`description`],
//     isPremium: false,
//     type: upperCaseFirst(`type1`),
//     rating: 4.8,
//     bedroom: 3,
//     visitor: 4,
//     price: 226,
//     option: [`option1`, `option2`, `option3`, `option4`],
//     isBookmark: false,
//     city: `city1`,
//     cityCoordinates: [48.85661, 2.351499],
//     cityZoom: 13,
//     host: `hostName`,
//     hostTop: false,
//     avatar: `avatar_url`,
//     coordinates: [48.877610000000004, 2.333499],
//   },
//   {
//     id: `4`,
//     img: [`url1`, `url2`, `url3`, `url4`, `url5`, `url6`],
//     smallImg: `small-url`,
//     title: `title`,
//     description: [`description`],
//     isPremium: false,
//     type: upperCaseFirst(`type1`),
//     rating: 4.8,
//     bedroom: 3,
//     visitor: 4,
//     price: 126,
//     option: [`option1`, `option2`, `option3`, `option4`],
//     isBookmark: false,
//     city: `city1`,
//     cityCoordinates: [48.85661, 2.351499],
//     cityZoom: 13,
//     host: `hostName`,
//     hostTop: false,
//     avatar: `avatar_url`,
//     coordinates: [48.877610000000004, 2.333499],
//   },
//   {
//     id: `7`,
//     img: [`url1`, `url2`, `url3`, `url4`, `url5`, `url6`],
//     smallImg: `small-url`,
//     title: `title`,
//     description: [`description`],
//     isPremium: false,
//     type: upperCaseFirst(`type1`),
//     rating: 4.8,
//     bedroom: 3,
//     visitor: 4,
//     price: 256,
//     option: [`option1`, `option2`, `option3`, `option4`],
//     isBookmark: false,
//     city: `city1`,
//     cityCoordinates: [48.85661, 2.351499],
//     cityZoom: 13,
//     host: `hostName`,
//     hostTop: false,
//     avatar: `avatar_url`,
//     coordinates: [48.877610000000004, 2.333499],
//   },
// ];
export const testInitialState = {
  COMMON: {
    offers: [],
    nearOffers: [],
    changedBookmarkOffer: {},
    pageOffer: {},
    bookmarkOffers: [],
    authInfo: {},
    cities: [`City1`, `City2`, `City3`, `City4`, `City5`, `City6`],
    currentCityName: `City1`,
    currentSortType: `Popular`,
    overOfferId: ``,
    reviews: [],
  },
  USER: {
    authorizationStatus: `NO_AUTH`,
  },
};
