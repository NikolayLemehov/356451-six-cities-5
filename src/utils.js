export const extend = (a, b) => {
  return Object.assign({}, a, b);
};

export const upperCaseFirst = (str) => str ? str[0].toUpperCase() + str.slice(1) : str;
