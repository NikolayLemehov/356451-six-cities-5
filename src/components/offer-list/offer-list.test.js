import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {MemoryRouter} from "react-router-dom";
import configureMockStore from "redux-mock-store";
import OfferList from "./offer-list";
import {testInitialState} from "../../test-data";
import {upperCaseFirst} from "../../utils";

const mockStore = configureMockStore();
const store = mockStore(testInitialState);

const offers = [
  {
    id: `1`,
    img: [`url1`, `url2`, `url3`, `url4`, `url5`, `url6`],
    smallImg: `small-url`,
    title: `title`,
    description: [`description`],
    isPremium: false,
    type: upperCaseFirst(`type1`),
    rating: 4.8,
    bedroom: 3,
    visitor: 4,
    price: 226,
    option: [`option1`, `option2`, `option3`, `option4`],
    isBookmark: false,
    city: `city1`,
    cityCoordinates: [48.85661, 2.351499],
    cityZoom: 13,
    host: `hostName`,
    hostTop: false,
    avatar: `avatar_url`,
    coordinates: [48.877610000000004, 2.333499],
  },
  {
    id: `4`,
    img: [`url1`, `url2`, `url3`, `url4`, `url5`, `url6`],
    smallImg: `small-url`,
    title: `title`,
    description: [`description`],
    isPremium: false,
    type: upperCaseFirst(`type1`),
    rating: 4.8,
    bedroom: 3,
    visitor: 4,
    price: 126,
    option: [`option1`, `option2`, `option3`, `option4`],
    isBookmark: false,
    city: `city1`,
    cityCoordinates: [48.85661, 2.351499],
    cityZoom: 13,
    host: `hostName`,
    hostTop: false,
    avatar: `avatar_url`,
    coordinates: [48.877610000000004, 2.333499],
  },
  {
    id: `7`,
    img: [`url1`, `url2`, `url3`, `url4`, `url5`, `url6`],
    smallImg: `small-url`,
    title: `title`,
    description: [`description`],
    isPremium: false,
    type: upperCaseFirst(`type1`),
    rating: 4.8,
    bedroom: 3,
    visitor: 4,
    price: 256,
    option: [`option1`, `option2`, `option3`, `option4`],
    isBookmark: false,
    city: `city1`,
    cityCoordinates: [48.85661, 2.351499],
    cityZoom: 13,
    host: `hostName`,
    hostTop: false,
    avatar: `avatar_url`,
    coordinates: [48.877610000000004, 2.333499],
  },
];

it(`OfferList should render correctly`, () => {
  const tree = renderer
    .create(
        <Provider store={store}>
          <MemoryRouter>
            <OfferList
              offers={offers}
              currentCardType={`main`}
            />
          </MemoryRouter>
        </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
