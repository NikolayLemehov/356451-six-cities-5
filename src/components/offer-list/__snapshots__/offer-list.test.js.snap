// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OfferList should render correctly 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <article
    className="
    cities__place-card
     place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="
       cities__image-wrapper
       place-card__image-wrapper"
    >
      <a
        className="place-card__img-link"
        href="/offer/1"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="small-url"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            226
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <a
          className="place-card__bookmark-button
         button"
          href="/login"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To
             bookmarks
          </span>
        </a>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "96%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          1
           
          title
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Type1
      </p>
    </div>
  </article>
  <article
    className="
    cities__place-card
     place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="
       cities__image-wrapper
       place-card__image-wrapper"
    >
      <a
        className="place-card__img-link"
        href="/offer/4"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="small-url"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            126
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <a
          className="place-card__bookmark-button
         button"
          href="/login"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To
             bookmarks
          </span>
        </a>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "96%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/4"
          onClick={[Function]}
        >
          4
           
          title
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Type1
      </p>
    </div>
  </article>
  <article
    className="
    cities__place-card
     place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="
       cities__image-wrapper
       place-card__image-wrapper"
    >
      <a
        className="place-card__img-link"
        href="/offer/7"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="small-url"
          width="260"
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            256
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <a
          className="place-card__bookmark-button
         button"
          href="/login"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To
             bookmarks
          </span>
        </a>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "96%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/7"
          onClick={[Function]}
        >
          7
           
          title
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        Type1
      </p>
    </div>
  </article>
</div>
`;
